sys/net/gnrc/sock/gnrc_sock.c

```diff
@@ -116,7 +119,14 @@ ssize_t gnrc_sock_recv(gnrc_sock_reg_t *reg, gnrc_pktsnip_t **pkt_out,
     if (mbox_size(&reg->mbox) != GNRC_SOCK_MBOX_SIZE) {
         return -EINVAL;
     }
-#ifdef MODULE_XTIMER
+#if IS_USED(MODULE_ZTIMER_USEC)
+    ztimer_t timeout_timer;
```

Comment:
This is a problem with `xtimer` as well, but this initialization might be with garbage in `base.next` ;-). Was about to provide a fix for that anyway any moment (just testing something before-hand), so I will just piggy-back an initializer here as well ;-)

Reply:
See https://github.com/RIOT-OS/RIOT/pull/17855